zypper.py

```diff
@@ -0,0 +1,274 @@
+#!/usr/bin/env python3
+
+"""
+Description: Expose metrics from zypper updates and patches.
+
+The script can take 2 arguments: `--more` and `--less`.
+The selection of the arguments change how many informations are going to be printed.
+
+The `--more` is by default.
+
+Examples:
+
+    zypper.py --less
+    zypper.py -m
+
+Authors: Gabriele Puliti <gabriele.puliti@suse.com>
+         Bernd Shubert <bschubert@suse.com>
+"""
+
+import argparse
+import subprocess
+import os
+import sys
+
+from collections.abc import Sequence
+from prometheus_client import CollectorRegistry, Gauge, Info, generate_latest
+
+REGISTRY = CollectorRegistry()
+NAMESPACE = "zypper"
+
+
+def __print_pending_data(data, fields, info, filters=None):
+    filters = filters or {}
+
+    if len(data) == 0:
+        field_str = ",".join([f'{name}=""' for _, name in fields])
+        info.info({field_str: '0'})
+    else:
+        for package in data:
+            check = all(package.get(k) == v for k, v in filters.items())
+            if check:
+                field_str = ",".join([f'{name}="{package[field]}"' for field, name in fields])
+                info.info({field_str: '1'})
+
+
+def print_pending_updates(data, all_info, filters=None):
+    if all_info:
+        fields = [("Repository", "repository"),
+                  ("Name", "package-name"),
+                  ("Available Version",
+                   "available-version")]
+    else:
+        fields = [("Repository", "repository"),
+                  ("Name", "package-name")]
+    prefix = "zypper_update_pending"
+    description = (
+        "zypper package update available from repository. "
+        "(0 = not available, 1 = available)"
+    )
+    info = Info(prefix, description)
```

Comment:
You specify a metric description that includes "(0 = not available, 1 = available)", yet `Info` metrics cannot take a user-defined value, and _always_ have a value of 1. I'm not sure what your intentions are here.

Reply:
@dswarbrick Thanks for your comments and effort to review it. The explanation for 0 and 1 in the help section clarifies what value can be expected to get from the metrics. Even if it is clear for some people not of of the users will have the same knowledge from the beginning. From my point of view having this hint mentioned here does hurt anyone and makes the usage a bit easier.

Reply:
@pirat013 The issue is not with having that information in the metric description. The issue is that using an `Info` metric type does not allow the **value** to be specified - they always have a value of 1. Encoding the 0 or 1 status as a **label** is not an acceptable approach, due to its propensity to create new series and result in stale metrics when that label value changes. Using a `Gauge` metric, and setting the metric **value** to 0 or 1, is the only correct way to do this.

Reply:
Thanks, got it.